lib/libutils/ext/include/util.h

```diff
@@ -188,7 +188,33 @@ static inline void reg_pair_from_64(uint64_t val, uint32_t *reg0,
 	*reg1 = low32_from_64(val);
 }
 
-/* Get and set bit fields  */
+/**
+ * Functions to get and set bit fields in a 32/64-bit register.
+ *
+ * These inline functions allow setting and extracting specific bits in
+ * a register @reg according to a given mask @mask. The mask
+ * specifies which bits in the register should be updated or extracted.
+ *
+ * - set_field*(): Modifies specific bits in a register by clearing the
```

Comment:
Maybe `set_field_u*()`.

No need to tell bits are cleared before being set, they are just set.
By the way, the implementation could be slightly improved:
```patch
-	return (reg & ~mask) | (val * (mask & ~(mask - 1)));
+	return reg | (val * (mask & ~(mask - 1)));
```

I think you should also tell that bits to be set are shifted to \@mask rightmost position.

Reply:
I don' t think your improvement might works. If you are removing the "& ~mask" the function will no longer "erease" the bits already present where the mask is placed.

example set 1010 on 1000011110000000 at position BIT(8) :

with the original :   1000010100000000
with your version :  1000011110000000

Reply:
Right, my mistake. Discard this part of my comment.